### PR TITLE
Add Spotify provider plus add missing env vars to docs

### PR DESCRIPTION
**Pull request by Yagger**

### CODE REVIEW

**Comment by elitan** [comment]
Thanks! I'm just waiting for this to merge: https://github.com/JMPerez/passport-spotify/pull/98

File: src/routes/auth/providers/spotify.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+export default (router: Router): void => {
+  const options = PROVIDERS.spotify
+  // Checks if the strategy is enabled. Don't create any route otherwise
+  if (options) {
+    // Checks if the strategy has at least a client ID and a client secret
+    if (!options.clientID || !options.clientSecret) {
+      throw Boom.badImplementation(`Missing environment variables for Spotify OAuth.`)
+    }
+    initProvider(router, 'spotify', Strategy, {
+        scope: ['user-read-email', 'user-read-private', 'app-remote-control', 'streaming']
```
Comment (elitan) [inline src/routes/auth/providers/spotify.ts:16]: I don't think wee need all these scopes by default. I think we can use `'user-read-email', 'user-read-private'` only.

If the app needs more access I think we should do that in a separate PR with: 

- scope per provider
- save access tokens to providers

**Review by elitan — CHANGES_REQUESTED** [review]

File: src/routes/auth/providers/spotify.ts (hunk 1 above)
Comment (Yagger) [reply src/routes/auth/providers/spotify.ts:16]: @elitan how about making provider scopes configurable via env vars? I could make separate PR for that.

**Comment by Yagger** [comment]
Removed extra scopes from spotify provider

File: src/routes/auth/providers/spotify.ts (hunk 1 above)
Comment (elitan) [reply src/routes/auth/providers/spotify.ts]: > @elitan how about making provider scopes configurable via env vars? I could make separate PR for that.

Ye. I think it make sense. The scope should be configurable for each provider.

Maybe you could open a new issue about it and we could take the discussion there? Some initial thoughts:

- With extra permission I think we want to store the OAuth token (https://github.com/nhost/hasura-backend-plus/pull/367). So that you could ex use the OAuth token to change song on Spotify?
- Do some provider rotate the OAuth token?

File: package.json [hunk 2]
```diff
@@ -63,6 +63,7 @@
     "passport-github2": "0.1.12",
     "passport-google-oauth20": "2.0.0",
     "passport-linkedin-oauth2": "^2.0.0",
+    "passport-spotify": "^1.1.0",
```
Comment (elitan) [inline package.json:66]: Let's update this to https://github.com/JMPerez/passport-spotify/releases/tag/v2.0.0

**Review by elitan — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (Yagger) [reply package.json]: This is done
